### DISCUSSION TITLE
How to enable AVX512VL and AVX512_VNNI without full AVX512?

### DISCUSSION THREAD

**Discussion by OpenSourceAnarchist** [ggml-org/llama.cpp#10349] [Q&A]
I'm customizing the build scripts for my local machines. Intel Cascade Lake - present all support AVX512VL and AVX512_VNNI instructions, but they don't all have full AVX512 support.

In the GitHub Actions build.yml:
```
- build: 'avx512-x64'
            defines: '-DGGML_NATIVE=OFF -DLLAMA_BUILD_SERVER=ON -DGGML_RPC=ON -DGGML_AVX512=ON -DBUILD_SHARED_LIBS=ON'
```
and later:
```
      - name: Check AVX512F support
        id: check_avx512f
        if: ${{ matrix.build == 'avx512-x64' }}
        continue-on-error: true
        run: |
          cd build
          $vcdir = $(vswhere -latest -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationPath)
          $msvc = $(join-path $vcdir $('VC\Tools\MSVC\'+$(gc -raw $(join-path $vcdir 'VC\Auxiliary\Build\Microsoft.VCToolsVersion.default.txt')).Trim()))
          $cl =  $(join-path $msvc 'bin\Hostx64\x64\cl.exe')
          echo 'int main(void){unsigned int a[4];__cpuid(a,7);return !(a[1]&65536);}' >> avx512f.c
          & $cl /O2 /GS- /kernel avx512f.c /link /nodefaultlib /entry:main
          .\avx512f.exe && echo "AVX512F: YES" && ( echo HAS_AVX512F=1 >> $env:GITHUB_ENV ) || echo "AVX512F: NO"
```
Is there a simple way to limit this and force just AVX512VL and AVX512_VNNI? I've never used CMake or even GH Actions before... perhaps this could be an enhancement/feature request if it isn't trivial. I appreciate any help or guidance, and thank you for giving inference to the masses <3

**Comment by clort81** [comment]
You can go into the build directory and use 'ccmake' command to set individual options, then 'c' to configure and 'g' to generate your modified cmake config.
